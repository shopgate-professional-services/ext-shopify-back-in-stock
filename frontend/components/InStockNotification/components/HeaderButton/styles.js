import { css } from 'glamor';
import {
  IN_STOCK_NOTIFICATION_BUTTON_COLOR,
  IN_STOCK_NOTIFICATION_BUTTON_TEXT_COLOR,
} from '../../../../constants';

const headerButton = css({
  width: '100%',
  display: 'block',
  padding: 3,
  textAlign: 'center',
  backgroundColor: IN_STOCK_NOTIFICATION_BUTTON_COLOR,
  color: IN_STOCK_NOTIFICATION_BUTTON_TEXT_COLOR,
  '&:focus': { outline: 0 },
});

export default { headerButton };
