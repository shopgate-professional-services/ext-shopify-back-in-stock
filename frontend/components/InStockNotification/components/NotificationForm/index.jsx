import React, { Component } from 'react';
import TextField from '@shopgate/pwa-ui-shared/TextField';

/**
 * Render Notification Form
 */
class NotificationForm extends Component {
  state = {
    emailIsValid: true,
    email: '',
  };

  componentDidMount() {
    const { email } = this.props;
    if (email && !this.state.email) {
      this.updateEmailValue(email);
    }
  }

  componentWillReceiveProps(nextProps) {
    const { email } = nextProps;
    if (email) {
      this.updateEmailValue(email);
    }
  }

  /**
   * Udpate email state
   * @param {string} email Entered email address
   */
  updateEmailValue = (email) => {
    this.setState({ email });
  };

  handelSubmit = (evemt) => {
    evemt.preventDefault();
    const emailIsValid = this.validateEmail(this.state.email);
    this.setState({ emailIsValid });

    if (!emailIsValid) {
      return;
    }
    const { productNumber, variantNumber, sendRequest } = this.props;
    sendRequest(productNumber, variantNumber, this.state.email);
  };

  validateEmail = email => /^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/.test(email);

  render() {
    const { isFetching } = this.props;
    return (
      <form onSubmit={this.handelSubmit}>
        <TextField
          name="notification-email"
          type="email"
          label="Email Address"
          onChange={this.updateEmailValue}
          value={this.state.email}
          errorText={this.state.emailIsValid ? '' : 'Please provide a valid email address'}
        />
        {!isFetching ?
          <button
            type="submit"
          >
          Submit
          </button> :
          <div>Fetching </div>
        }
      </form>
    );
  }
}

export default NotificationForm;
