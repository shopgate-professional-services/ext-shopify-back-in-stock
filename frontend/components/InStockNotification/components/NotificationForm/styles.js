import { css } from 'glamor';
import {
  IN_STOCK_NOTIFICATION_BUTTON_COLOR,
  IN_STOCK_NOTIFICATION_BUTTON_TEXT_COLOR,
} from '../../../../constants';

const overallWrapper = css({
  padding: 3,
}).toString();

const submitButton = css({
  backgroundColor: IN_STOCK_NOTIFICATION_BUTTON_COLOR,
  color: `${IN_STOCK_NOTIFICATION_BUTTON_TEXT_COLOR} !important`,
}).toString();

const formWrapperBase = {
  overflow: 'hidden',
  width: '100%',
  padding: 3,
  transition: 'height .3s',
};

const closedFormWrapper = css({
  ...formWrapperBase,
  height: 0,
}).toString();

const openFormWrapper = css({
  ...formWrapperBase,
  height: 150,
}).toString();

export default {
  closedFormWrapper,
  openFormWrapper,
  submitButton,
  overallWrapper,
};
