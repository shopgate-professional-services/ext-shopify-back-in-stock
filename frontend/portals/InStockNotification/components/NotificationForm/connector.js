import { connect } from 'react-redux';
import { getUserEmail } from '@shopgate/pwa-common/selectors/user';

/**
 * Maps the contents of the state to the component props.
 * @param {Object} state The current application state.
 * @return {Object} The extended component props.
 */
const mapStateToProps = state => ({
  email: getUserEmail(state),
});

export default connect(mapStateToProps);
