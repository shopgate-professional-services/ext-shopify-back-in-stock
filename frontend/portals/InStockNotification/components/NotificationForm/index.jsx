import React, { Component } from 'react';
import TextField from '@shopgate/pwa-ui-shared/TextField';
import connect from './connector';

/**
 * Render Notification Form
 */
class NotificationForm extends Component {
  state = {
    emailIsValid: true,
    email: '',
  };

  componentDidMount() {
    const { email } = this.props;
    if (email && !this.state.email) {
      this.updateEmailValue(email);
    }
  }

  componentWillReceiveProps(nextProps) {
    const { email } = nextProps;
    if (email) {
      this.updateEmailValue(email);
    }
  }

  /**
   * Udpate email state
   * @param {string} email Entered email address
   */
  updateEmailValue = (email) => {
    this.setState({ email });
  };

  handelSubmit = (evemt) => {
    evemt.preventDefault();
    const emailIsValid = this.validateEmail(this.state.email);
    this.setState({ emailIsValid });

    if (emailIsValid) {
      console.log('form submit');
    }
  };

  validateEmail = email => /^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/.test(email);

  render() {
    return (
      <form onSubmit={this.handelSubmit}>
        <TextField
          name="notification-email"
          type="email"
          label="Email Address"
          onChange={this.updateEmailValue}
          value={this.state.email}
          errorText={this.state.emailIsValid ? '' : 'Please provide a valid email address'}
        />
        <button
          type="submit"
        >
          Submit
        </button>
      </form>
    );
  }
}

export default connect(NotificationForm);
