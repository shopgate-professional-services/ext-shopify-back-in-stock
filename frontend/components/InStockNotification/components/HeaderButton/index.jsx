import React from 'react';
import PropTypes from 'prop-types';
import I18n from '@shopgate/pwa-common/components/I18n';
import styles from './styles';

/**
 * Render HeaderButton
 * @param {string} text Button text
 * @param {Function} onClick Function to call when clicked
 * @return {JSX}
 */
const HeaderButton = ({ text, onClick }) => (
  <button
    onClick={onClick}
    className={styles.headerButton}
  >
    <div>
      <I18n.Text string={text} />
    </div>
  </button>
);

HeaderButton.propTypes = {
  text: PropTypes.string.isRequired,
  onClick: PropTypes.func,
};

HeaderButton.defaultProps = {
  onClick: () => {},
};

export default HeaderButton;
