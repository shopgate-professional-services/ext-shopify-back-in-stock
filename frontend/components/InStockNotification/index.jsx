import React, { Component } from 'react';
import NotificationForm from './components/NotificationForm';

class InstockNotification extends Component {
  render() {
    const {
      showForm,
      productNumber,
      variantNumber,
      email,
      confirmationIsFetching,
      confirmationMessage,
      sendRequest,
    } = this.props;
    if (confirmationMessage) {
      return <div>{confirmationMessage}</div>;
    }
    if (showForm && productNumber) {
      return (
        <NotificationForm
          productNumber={productNumber}
          variantNumber={variantNumber}
          email={email}
          isFetching={confirmationIsFetching}
          sendRequest={sendRequest}
        />);
    }

    return <div>hide form</div>;
  }
}

export default InstockNotification;
