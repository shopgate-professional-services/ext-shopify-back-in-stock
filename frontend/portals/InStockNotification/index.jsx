import React from 'react';
import { RouteContext } from '@shopgate/pwa-common/context';
import { hex2bin } from '@shopgate/pwa-common/helpers/data';
import connect from './connector';
import NotificationForm from './components/NotificationForm';

const InStockNotification = connect(({ productNumber, variantNumber, showForm}) => {
  return showForm && productNumber ? <NotificationForm /> : <div>hide form</div>;
});

export default props => (
  <RouteContext.Consumer>
    {({ params }) => (
      <div>
        <InStockNotification {...props} productId={hex2bin(params.productId)} />
      </div>
    )}
  </RouteContext.Consumer>
);
