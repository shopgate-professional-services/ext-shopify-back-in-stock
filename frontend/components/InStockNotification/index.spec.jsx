import React from 'react';
import { mount } from 'enzyme';
import InStockNotification from './index';
import HeaderButton from './components/HeaderButton';
import NotificationForm from './components/NotificationForm';

const GENERAL_PROPS = {
  showForm: true,
  confirmationMessage: null,
  productNumber: 'testProductNumber',
  variantNumber: 'testVariantNumber',
  email: 'someone@someplace.com',
  isFetching: false,
  sendRequest: () => {},
};

describe('inStockNotification', () => {
  it('Should return null because showForm is false', () => {
    const props = {
      productNumber: 'testProduct',
      showForm: false,
    };
    const component = mount(<InStockNotification {...props} />);
    expect(component.html()).toBe(null);
  });
  it('Should return null because there is no productNumber', () => {
    const props = {
      productNumber: null,
      showForm: true,
    };
    const component = mount(<InStockNotification {...props} />);
    expect(component.html()).toBe(null);
  });
  it('should show only HeaderButton because there is a confirmation notice', () => {
    const component = mount(<InStockNotification confirmationMessage="some message" />);
    expect(component.find(HeaderButton).length).toBe(1);
    expect(component.find(NotificationForm).length).toBe(0);
    expect(component).toMatchSnapshot();
  });
  it('should show notification form because there is a productNumber, showForm is true, and confirmationMessage is null', () => {
    const component = mount(<InStockNotification {...GENERAL_PROPS} />);
    expect(component.find(HeaderButton).length).toBe(1);
    expect(component.find(NotificationForm).length).toBe(1);
    expect(component).toMatchSnapshot();
  });
  it('should show notification while fetching', () => {
    const props = {
      ...GENERAL_PROPS,
      isFetching: true,
    };
    const component = mount(<InStockNotification {...props} />);
    expect(component.find(HeaderButton).length).toBe(1);
    expect(component.find(NotificationForm).length).toBe(1);
    expect(component).toMatchSnapshot();
  });
});
