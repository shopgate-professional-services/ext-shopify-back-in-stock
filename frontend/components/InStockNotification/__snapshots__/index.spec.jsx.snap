// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`inStockNotification should show notification form because there is a productNumber, showForm is true, and confirmationMessage is null 1`] = `
<InStockNotification
  confirmationIsFetching={false}
  confirmationMessage={null}
  email="someone@someplace.com"
  isFetching={false}
  productNumber="testProductNumber"
  sendRequest={[Function]}
  showConfirmationModal={[Function]}
  showForm={true}
  variantNumber="testVariantNumber"
>
  <NotificationForm
    email="someone@someplace.com"
    isFetching={false}
    productNumber="testProductNumber"
    sendRequest={[Function]}
    variantNumber="testVariantNumber"
  >
    <div
      className="css-1q1mvko"
    >
      <HeaderButton
        onClick={[Function]}
        text="inStockNotification.header.before_submit"
      >
        <button
          className={
            Object {
              "data-css-wmlzrp": "",
            }
          }
          onClick={[Function]}
        >
          <div>
            <Translate
              className=""
              params={Object {}}
              string="inStockNotification.header.before_submit"
            >
              <span
                className=""
              >
                inStockNotification.header.before_submit
              </span>
            </Translate>
          </div>
        </button>
      </HeaderButton>
      <div
        className="css-159u6xo"
      >
        <Form
          onSubmit={[Function]}
        >
          <form
            action="#"
            onKeyPress={[Function]}
            onSubmit={[Function]}
          >
            <TextField
              className=""
              errorText=""
              hintText=""
              isControlled={false}
              label="inStockNotification.form.input.label"
              multiLine={false}
              name="notification-email"
              onChange={[Function]}
              onFocusChange={[Function]}
              onSanitize={[Function]}
              onValidate={[Function]}
              password={false}
              setRef={[Function]}
              translateErrorText={true}
              type="email"
              value="someone@someplace.com"
            >
              <div
                className="css-e4z0u6 "
              >
                <Hint
                  hintText=""
                  visible={false}
                >
                  <div
                    className="css-tnfb18 css-1jyyyen"
                  >
                    <Translate
                      className=""
                      params={Object {}}
                      string=""
                    />
                  </div>
                </Hint>
                <Label
                  hasErrorMessage={false}
                  isFloating={true}
                  isFocused={false}
                  label="inStockNotification.form.input.label"
                  name="notification-email"
                >
                  <label
                    className="css-u7lqu6 css-16msjt6 css-1fekhao"
                    htmlFor="notification-email"
                  >
                    <Translate
                      className=""
                      params={Object {}}
                      string="inStockNotification.form.input.label"
                    >
                      <span
                        className=""
                      >
                        inStockNotification.form.input.label
                      </span>
                    </Translate>
                  </label>
                </Label>
                <FormElement
                  id="notification-email"
                  isControlled={false}
                  multiLine={false}
                  name="notification-email"
                  onChange={[Function]}
                  onFocusChange={[Function]}
                  onSanitize={[Function]}
                  onValidate={[Function]}
                  password={false}
                  setRef={[Function]}
                  type="email"
                  value="someone@someplace.com"
                >
                  <Component
                    className="css-keangt"
                    id="notification-email"
                    isControlled={false}
                    multiLine={false}
                    name="notification-email"
                    onChange={[Function]}
                    onFocusChange={[Function]}
                    onSanitize={[Function]}
                    onValidate={[Function]}
                    password={false}
                    setRef={[Function]}
                    type="email"
                    validateOnBlur={true}
                    value="someone@someplace.com"
                  >
                    <SimpleInput
                      autoComplete={false}
                      autoCorrect={false}
                      className="css-keangt"
                      disabled={false}
                      id="notification-email"
                      isControlled={false}
                      multiLine={false}
                      name="notification-email"
                      onChange={[Function]}
                      onFocusChange={[Function]}
                      onKeyPress={[Function]}
                      onSanitize={[Function]}
                      onValidate={[Function]}
                      password={false}
                      setRef={[Function]}
                      type="email"
                      validateOnBlur={true}
                      value="someone@someplace.com"
                    >
                      <input
                        autoComplete="off"
                        autoCorrect="off"
                        className="css-keangt"
                        disabled={false}
                        id="notification-email"
                        name="notification-email"
                        onBlur={[Function]}
                        onChange={[Function]}
                        onFocus={[Function]}
                        onKeyPress={[Function]}
                        type="email"
                        value="someone@someplace.com"
                      />
                    </SimpleInput>
                  </Component>
                </FormElement>
                <Underline
                  hasErrorMessage={false}
                  isFocused={false}
                >
                  <div
                    className="css-ycal4l"
                  >
                    <div
                      className="css-1tpmuef"
                      style={
                        Object {
                          "borderBottomColor": undefined,
                          "transform": "scale3d(0,1,1)",
                        }
                      }
                    />
                  </div>
                </Underline>
                <ErrorText
                  className=""
                  errorText=""
                  translate={true}
                  validationError={null}
                >
                  <div
                    className=" css-hmgfge"
                  >
                    <Translate
                      className=""
                      params={Object {}}
                      string=""
                    />
                  </div>
                </ErrorText>
              </div>
            </TextField>
            <ActionButton
              className="css-1a5gmoq"
              disabled={false}
              flat={true}
              loading={false}
              noGap={false}
              onClick={[Function]}
              rippleClassName=""
              rippleSize={null}
              testId={null}
              type="secondary"
              wrapContent={true}
            >
              <div
                className="css-omcrdt"
                data-test-id={null}
              >
                <RippleButton
                  className="css-1a5gmoq"
                  disabled={false}
                  flat={true}
                  onClick={[Function]}
                  rippleClassName=""
                  rippleSize={null}
                  testId="Button"
                  type="secondary"
                  wrapContent={true}
                >
                  <Button
                    className="css-1a5gmoq"
                    disabled={false}
                    flat={true}
                    onClick={[Function]}
                    testId="Button"
                    type="secondary"
                    wrapContent={false}
                  >
                    <Button
                      className="css-dh0pju css-1a5gmoq"
                      disabled={false}
                      onClick={[Function]}
                      testId="Button"
                    >
                      <button
                        className="css-dh0pju css-1a5gmoq css-1rn65c0"
                        data-test-id="Button"
                        disabled={false}
                        onClick={[Function]}
                      >
                        <Ripple
                          className="css-1fscwua "
                          color="#000"
                          disabled={false}
                          fill={true}
                          onClick={[Function]}
                          onComplete={[Function]}
                          overflow={false}
                          size={null}
                        >
                          <div
                            aria-hidden={true}
                            className="css-1fscwua "
                            data-test-id="Ripple"
                            onClick={[Function]}
                            style={
                              Object {
                                "overflow": "hidden",
                                "position": "relative",
                              }
                            }
                          >
                            <Translate
                              className=""
                              params={Object {}}
                              string="inStockNotification.form.submit"
                            >
                              <span
                                className=""
                              >
                                inStockNotification.form.submit
                              </span>
                            </Translate>
                          </div>
                        </Ripple>
                      </button>
                    </Button>
                  </Button>
                </RippleButton>
              </div>
            </ActionButton>
          </form>
        </Form>
      </div>
    </div>
  </NotificationForm>
</InStockNotification>
`;

exports[`inStockNotification should show notification while fetching 1`] = `
<InStockNotification
  confirmationIsFetching={false}
  confirmationMessage={null}
  email="someone@someplace.com"
  isFetching={true}
  productNumber="testProductNumber"
  sendRequest={[Function]}
  showConfirmationModal={[Function]}
  showForm={true}
  variantNumber="testVariantNumber"
>
  <NotificationForm
    email="someone@someplace.com"
    isFetching={false}
    productNumber="testProductNumber"
    sendRequest={[Function]}
    variantNumber="testVariantNumber"
  >
    <div
      className="css-1q1mvko"
    >
      <HeaderButton
        onClick={[Function]}
        text="inStockNotification.header.before_submit"
      >
        <button
          className={
            Object {
              "data-css-wmlzrp": "",
            }
          }
          onClick={[Function]}
        >
          <div>
            <Translate
              className=""
              params={Object {}}
              string="inStockNotification.header.before_submit"
            >
              <span
                className=""
              >
                inStockNotification.header.before_submit
              </span>
            </Translate>
          </div>
        </button>
      </HeaderButton>
      <div
        className="css-159u6xo"
      >
        <Form
          onSubmit={[Function]}
        >
          <form
            action="#"
            onKeyPress={[Function]}
            onSubmit={[Function]}
          >
            <TextField
              className=""
              errorText=""
              hintText=""
              isControlled={false}
              label="inStockNotification.form.input.label"
              multiLine={false}
              name="notification-email"
              onChange={[Function]}
              onFocusChange={[Function]}
              onSanitize={[Function]}
              onValidate={[Function]}
              password={false}
              setRef={[Function]}
              translateErrorText={true}
              type="email"
              value="someone@someplace.com"
            >
              <div
                className="css-e4z0u6 "
              >
                <Hint
                  hintText=""
                  visible={false}
                >
                  <div
                    className="css-tnfb18 css-1jyyyen"
                  >
                    <Translate
                      className=""
                      params={Object {}}
                      string=""
                    />
                  </div>
                </Hint>
                <Label
                  hasErrorMessage={false}
                  isFloating={true}
                  isFocused={false}
                  label="inStockNotification.form.input.label"
                  name="notification-email"
                >
                  <label
                    className="css-u7lqu6 css-16msjt6 css-1fekhao"
                    htmlFor="notification-email"
                  >
                    <Translate
                      className=""
                      params={Object {}}
                      string="inStockNotification.form.input.label"
                    >
                      <span
                        className=""
                      >
                        inStockNotification.form.input.label
                      </span>
                    </Translate>
                  </label>
                </Label>
                <FormElement
                  id="notification-email"
                  isControlled={false}
                  multiLine={false}
                  name="notification-email"
                  onChange={[Function]}
                  onFocusChange={[Function]}
                  onSanitize={[Function]}
                  onValidate={[Function]}
                  password={false}
                  setRef={[Function]}
                  type="email"
                  value="someone@someplace.com"
                >
                  <Component
                    className="css-keangt"
                    id="notification-email"
                    isControlled={false}
                    multiLine={false}
                    name="notification-email"
                    onChange={[Function]}
                    onFocusChange={[Function]}
                    onSanitize={[Function]}
                    onValidate={[Function]}
                    password={false}
                    setRef={[Function]}
                    type="email"
                    validateOnBlur={true}
                    value="someone@someplace.com"
                  >
                    <SimpleInput
                      autoComplete={false}
                      autoCorrect={false}
                      className="css-keangt"
                      disabled={false}
                      id="notification-email"
                      isControlled={false}
                      multiLine={false}
                      name="notification-email"
                      onChange={[Function]}
                      onFocusChange={[Function]}
                      onKeyPress={[Function]}
                      onSanitize={[Function]}
                      onValidate={[Function]}
                      password={false}
                      setRef={[Function]}
                      type="email"
                      validateOnBlur={true}
                      value="someone@someplace.com"
                    >
                      <input
                        autoComplete="off"
                        autoCorrect="off"
                        className="css-keangt"
                        disabled={false}
                        id="notification-email"
                        name="notification-email"
                        onBlur={[Function]}
                        onChange={[Function]}
                        onFocus={[Function]}
                        onKeyPress={[Function]}
                        type="email"
                        value="someone@someplace.com"
                      />
                    </SimpleInput>
                  </Component>
                </FormElement>
                <Underline
                  hasErrorMessage={false}
                  isFocused={false}
                >
                  <div
                    className="css-ycal4l"
                  >
                    <div
                      className="css-1tpmuef"
                      style={
                        Object {
                          "borderBottomColor": undefined,
                          "transform": "scale3d(0,1,1)",
                        }
                      }
                    />
                  </div>
                </Underline>
                <ErrorText
                  className=""
                  errorText=""
                  translate={true}
                  validationError={null}
                >
                  <div
                    className=" css-hmgfge"
                  >
                    <Translate
                      className=""
                      params={Object {}}
                      string=""
                    />
                  </div>
                </ErrorText>
              </div>
            </TextField>
            <ActionButton
              className="css-1a5gmoq"
              disabled={false}
              flat={true}
              loading={false}
              noGap={false}
              onClick={[Function]}
              rippleClassName=""
              rippleSize={null}
              testId={null}
              type="secondary"
              wrapContent={true}
            >
              <div
                className="css-omcrdt"
                data-test-id={null}
              >
                <RippleButton
                  className="css-1a5gmoq"
                  disabled={false}
                  flat={true}
                  onClick={[Function]}
                  rippleClassName=""
                  rippleSize={null}
                  testId="Button"
                  type="secondary"
                  wrapContent={true}
                >
                  <Button
                    className="css-1a5gmoq"
                    disabled={false}
                    flat={true}
                    onClick={[Function]}
                    testId="Button"
                    type="secondary"
                    wrapContent={false}
                  >
                    <Button
                      className="css-dh0pju css-1a5gmoq"
                      disabled={false}
                      onClick={[Function]}
                      testId="Button"
                    >
                      <button
                        className="css-dh0pju css-1a5gmoq css-1rn65c0"
                        data-test-id="Button"
                        disabled={false}
                        onClick={[Function]}
                      >
                        <Ripple
                          className="css-1fscwua "
                          color="#000"
                          disabled={false}
                          fill={true}
                          onClick={[Function]}
                          onComplete={[Function]}
                          overflow={false}
                          size={null}
                        >
                          <div
                            aria-hidden={true}
                            className="css-1fscwua "
                            data-test-id="Ripple"
                            onClick={[Function]}
                            style={
                              Object {
                                "overflow": "hidden",
                                "position": "relative",
                              }
                            }
                          >
                            <Translate
                              className=""
                              params={Object {}}
                              string="inStockNotification.form.submit"
                            >
                              <span
                                className=""
                              >
                                inStockNotification.form.submit
                              </span>
                            </Translate>
                          </div>
                        </Ripple>
                      </button>
                    </Button>
                  </Button>
                </RippleButton>
              </div>
            </ActionButton>
          </form>
        </Form>
      </div>
    </div>
  </NotificationForm>
</InStockNotification>
`;

exports[`inStockNotification should show only HeaderButton because there is a confirmation notice 1`] = `
<InStockNotification
  confirmationIsFetching={false}
  confirmationMessage="some message"
  email={null}
  productNumber={null}
  sendRequest={[Function]}
  showConfirmationModal={[Function]}
  showForm={false}
  variantNumber={null}
>
  <HeaderButton
    onClick={[Function]}
    text="inStockNotification.header.after_submit"
  >
    <button
      className={
        Object {
          "data-css-wmlzrp": "",
        }
      }
      onClick={[Function]}
    >
      <div>
        <Translate
          className=""
          params={Object {}}
          string="inStockNotification.header.after_submit"
        >
          <span
            className=""
          >
            inStockNotification.header.after_submit
          </span>
        </Translate>
      </div>
    </button>
  </HeaderButton>
</InStockNotification>
`;
